mlflow/tracking/fluent.py

```diff
@@ -237,6 +318,14 @@ def delete_tag(key):
     will create a new active run.
 
     :param key: Name of the tag
+
+    .. code-block:: python
+        :caption: Example
+
+        import mlflow
+
+        # Remove a tag from this run
+        mlflow.delete_tag("engineering_remote")
```

Comment:
```python
import mlflow

with mlflow.start_run() as run:
    mlflow.set_tags({"foo": 0, "bar": 1})

with mlflow.start_run(run_id=run.info.run_id):
    mlflow.delete_tag("foo")
```

because the current code raise the following error?

```
mlflow.exceptions.MlflowException: No tag with name: engineering_remote in run with id xxx
```

Reply:
Yes, this makes more sense to delete a tag after it has been set in the previous run and start a new run with its previous id. Consider it done

```
tags = {"engineering": "ML Platform",
            "engineering_remote": "ML Platform"}
    
    with mlflow.start_run() as run:
        mlflow.set_tags(tags)

    with mlflow.start_run(run_id=run.info.run_id):
        mlflow.delete_tag("engineering_remote")
```